File: src/main/java/com/sem/keeper/repo/DeviceRepository.java
package com.sem.keeper.repo;

import com.sem.keeper.entity.DeviceEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * Eszköz entitások elérésére használó repository
 */
@Repository
public interface DeviceRepository extends PagingAndSortingRepository<DeviceEntity, Long> {

    DeviceEntity findById(long id);

    Page<DeviceEntity> findByNameContainingIgnoreCase(String name, Pageable page);
}

File: src/main/java/com/sem/keeper/repo/DeviceRepository.kt
package com.sem.keeper.repo

import com.sem.keeper.entity.DeviceEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface DeviceRepository : PagingAndSortingRepository<DeviceEntity, Long> {
    fun findByNameContainingIgnoreCase(name: String?, page: Pageable?): Page<DeviceEntity?>
}